Asterix/optics/propagation_functions.py

```diff
@@ -585,55 +586,98 @@ def prop_fpm_regional_sampling(pup,
-    samp_outer : float
-        Sampling in the outermost layer of propagations.
+        Shift of FPM with respect to optical axis in units of lambda/D.
     filter_order : int
         Order of the Butterworth filter.
     alpha : float
         Scale factor for the filter size. The larger this number, the smaller the filter size with respect to the
         input array.
+    dir_save_all_planes : string, default None
+        Directory to save all planes into fits files if save_all_planes_to_fits=True.
 
     Returns
     -------
     EF_before_LS : 2D array (complex)
         E-field before the Lyot stop.
     """
-    from Asterix.optics import shift_phase_ramp
+    samp_outer = 2
 
-    if samp_outer != 2:
-        raise ValueError(f"samp_outer' needs to be 2, otherwise we cut off the high-spatial frequencies and the"
-                         f"simulation turns out bad. This can become a variable parameter in the future if we allow"
-                         f"the array sizes to be adapted in this function as well, in order to retain the original"
-                         f"spatial frequency content in the outer sampling layer.")
+    if not isinstance(nbres, list):
+        raise TypeError(f"'nbres' parameter needs to be of type list or numpy.array. Currently type = {type(nbres)}")
 
-    dim_pup = pup.shape[0]
+    nbres = np.array(nbres)
+
+    dim_overpad_pupil = pup.shape[0]
     dim_fpm = fpm.shape[0]
-    pup0 = np.array(pup, copy=True, dtype='complex128')
 
-    # Add tip-tilt to in pupil wavefront to simulate FPM offsets
-    phase_ramp = shift_phase_ramp(dim_pup, shift[0], shift[1])
-    inverted_phase_ramp = shift_phase_ramp(dim_pup, -shift[0], -shift[1])
+    if real_dim_input is None:
+        real_dim_input = dim_overpad_pupil
+    samplings = real_dim_input / nbres
+
+    if not np.all(np.diff(nbres) >= 0):
+        raise ValueError(f"'nbres' parameter needs to be sorted from the highest to lowest number of elements."
+                         f"Currently 'nbres' = {nbres}")
+
+    if np.min(samplings) < 2:
+        raise ValueError(f"The outer sampling in prop_fpm_regional_sampling is hardcoded to 2, otherwise we cut off"
+                         f"the high-spatial frequencies and the simulation turns out bad. We need the samplings"
+                         f"defined by the 'nbres' parameter (real_dim_input/nbres) to be always >= 2. Currently, with"
+                         f"real_dim_input = {real_dim_input}, samplings are {samplings}.")
+
+    if np.min(samplings) == 2:
+        # The outer sampling defined by nbrs is already 2 (samp_outer = 2). In the case the maximum element in 'nbrs'
+        # correspond to a sampling of 2, we can remove it and gain some time.
+        nbres = np.delete(nbres, -1)
+        samplings = np.delete(samplings, -1)
+
+    if max(shift) >= nbres[0] / 2:
+        raise ValueError(f"shift {shift} is larger than the minimum nbrs {nbres[0]} of element of resolution : the "
```

Comment:
Again `nbrs` instead of the variable we actually use, `nbres`. Be careful with these kind of things, it can be very, very confusing. Could you fix it here too? And maybe do a "search for all" in this file and check it's not happening somewhere else as well.

Reply:
done